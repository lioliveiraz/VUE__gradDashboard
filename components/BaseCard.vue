  
<template>
  <div class="w-full card--main">
    <div>
      <h4 data-testId="graduate_name" class="text-orange-100 text-6xl">
        {{ graduated.name }}
      </h4>

      <p class="text-gray-100" data-testId="graduate_empID">
        Employeer id: {{ graduated.empId }}
      </p>
    </div>
  </div>
</template>

    <script>
export default {
  props: {
    graduated: Object,
  },
};
</script>
<style scoped>
.card--main {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  overflow: hidden;
}
.card--main div {
  width: 80%;
  height: 80%;
  text-transform: capitalize;
  cursor: pointer;
}
.card--main:hover {
  box-shadow: inset 1px 1px 100px #00c300;
}
</style>