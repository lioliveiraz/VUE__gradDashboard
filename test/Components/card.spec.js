import BaseCard from '../../components/BaseCard.vue';
import { shallowMount, mount } from '@vue/test-utils';
import { createLocalVue } from "@vue/test-utils";

const localVue = createLocalVue();
localVue.directive("t", () => { });



describe('<BaseCard/>', () => {
    let graduated, wrapper;
    beforeEach(async () => {
        graduated = {
            name: "name",
            empId: "123456"
        };
        wrapper = await shallowMount(BaseCard, { propsData: { graduated }, localVue });
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('graduate_name')).toBeTruthy();
        expect(wrapper.find('graduate_empID')).toBeTruthy();

    });
    it('should receive props correctly', () => {
        expect(wrapper.props()).toEqual({ graduated });
        expect(wrapper.text()).toContain("name");
    });

});