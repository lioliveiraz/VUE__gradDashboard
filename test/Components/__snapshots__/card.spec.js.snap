// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseCard/> should render correctly 1`] = `undefined`;
